### ISSUE TITLE
Reward functions for GymEnvironment: composite signal per env

### ISSUE THREAD

**Issue by mercurialsolo** [mercurialsolo/mantis#68]
## Problem

`GymEnvironment.step()` already returns `GymResult(observation, reward, done, info)`, but every adapter (`PlaywrightGymEnv`, `GymAnythingAdapter`, `XdotoolGymEnv`) currently returns `reward=0.0`. Without a real signal we cannot do RL fine-tuning (#TBD-finetuning) and we cannot grade rollouts for rejection sampling.

We also already have most of the *raw signals* we'd need scattered across the codebase — they just aren't normalized into a reward:
- `dynamic_plan_verifier.py` / `step_verifier.py` — per-step plan adherence
- `gym_result.info`: `backtracked`, `type_verified`, `focused_input`, `url`, `title`
- BoatTrader gate predicate (record completeness)
- OSWorld per-task evaluator (`evaluation_examples/`)
- VWA evaluator (`benchmarks/visualwebarena.py`)

## Proposed solution

Create `src/mantis_agent/rewards/` with one module per environment, each exporting a `compute_reward(trajectory_step, episode_state) -> float` function plus an `episode_reward(run_result) -> float`.

Recommended composite shape (weights tuned per env, **not** summed blindly):

```
r_step  = 0.3 * step_verified            # plan adherence
        + 0.1 * format_valid             # action parses cleanly
        - 0.5 * off_site_navigation
        - 0.2 * action_loop_detected

r_final = 1.0 * task_success             # binary, terminal
        + 0.3 * fraction_plan_steps_verified
```

Tuned negatives matter — without them GRPO finds reward hacks (gallery escapes, off-site clicks) within ~2K rollouts.

### Per-environment graders

- `rewards/boattrader.py` — gate predicate: required fields present + types correct + URL matches search filters
- `rewards/osworld.py` — wrap the existing OSWorld evaluator script as a terminal reward
- `rewards/vwa.py` — wrap VWA `evaluate()` as a terminal reward
- `rewards/plan_adherence.py` — generic, derives from `dynamic_plan_verifier`

### Wire-up

1. Adapters call the appropriate grader from `step()` and populate `GymResult.reward`.
2. `GymRunner` already accumulates `total_reward` and persists per-step reward into `TrajectoryStep` — no runner change needed.
3. Add a `--reward-fn` flag to existing run scripts so we can A/B reward shapes without code changes.

## Scope

Start with `rewards/plan_adherence.py` + `rewards/boattrader.py` since BoatTrader has a known-good gate and 35 leads of validation data already on disk (see auto-memory).

## Priority

**P2-medium** — prerequisite for fine-tuning. Cheap to land (≤500 LoC) and immediately useful for grading rollouts even before any training.
